<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>

<body>
  <script>
    async function main() {
      let p = new Promise((resolve, reject) => {
        resolve('ok')
        // reject("errorrrrr")
      })
      let p2 = new Promise((resolve, reject) => {
        // resolve('ok')
        reject("errorrrrr")
      })
      // 如果promise成功 则 await返回成功的值
      let res = await p
      // 如果右侧不是promise 则返回右侧的值
      let res2 = await 530
      // 如果是失败的promise 会抛错 用try catch来处理
      try {
        let res3 = await p2
      } catch (e) {
        console.warn(e)
      }
      console.log(res)
      console.log(res2)

    }
    main()
  </script>
</body>

</html>